engine/utils.hpp

```diff
@@ -117,6 +117,41 @@ static inline bool equal_string_view(const StringView &src,
   return false;
 }
 
+template <typename T> class FixedVector {
+public:
+  FixedVector(uint64_t size) : size_(size) { data_ = new T[size]; }
```

Comment:
explicit

Reply:
No valid object is constructed.
Assigning to reference got from operator[] is UB.